components/library/DateRangePicker/Preview/DateRangePicker.jsx

```diff
@@ -0,0 +1,214 @@
+  const today = new Date(new Date().setHours(0, 0, 0, 0));
+  const yesterday = new Date(
+    new Date(new Date().setDate(today.getDate() - 1)).setHours(0, 0, 0, 0)
+  );
+
+  const ranges = [
+    { title: 'Today', period: { since: today, until: today } },
+    { title: 'Yesterday', period: { since: yesterday, until: yesterday } },
+    {
+      title: 'Last 7 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 7)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 30 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 30)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 90 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 90)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 365 Days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 365)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Custom',
+      period: { since: yesterday, until: yesterday }
+    }
+  ];
+
+  const [popoverActive, setPopoverActive] = useState(false);
+  const [activeDateRange, setActiveDateRange] = useState(ranges[0]);
+  const [dateState, setDateState] = useState({
+    month: activeDateRange.period.since.getMonth(),
+    year: activeDateRange.period.since.getFullYear()
+  });
+
+  useEffect(() => {
+    setDateState({
+      month: activeDateRange.period.since.getMonth(),
+      year: activeDateRange.period.since.getFullYear()
+    });
+  }, [activeDateRange]);
+
+  const handleInputValueChange = useCallback((field, value) => {
+    if (isValidDate(value)) {
+      const newDate = new Date(value);
+      setActiveDateRange((prevState) => ({
+        ...prevState,
+        period: {
+          ...prevState.period,
+          [field]: newDate
+        }
+      }));
+    }
+  }, []);
+
+  function formatDate(date) {
+    return date.toISOString().split('T')[0];
+  }
+
+  function isValidDate(date) {
+    return !isNaN(new Date(date).getDate());
+  }
+
+  return (
+    <Box>
+      <Popover
+        active={popoverActive}
+        autofocusTarget='none'
+        preferredAlignment='left'
+        preferredPosition='below'
+        fluidContent
+        sectioned={false}
+        fullHeight
+        activator={
+          <Button icon={CalendarIcon} size='slim' onClick={() => setPopoverActive(!popoverActive)}>
+            {activeDateRange.title}
+          </Button>
+        }
+        onClose={() => setPopoverActive(false)}
+      >
+        <Box
+          style={{
+            paddingTop: '16px',
+            paddingLeft: '10px',
+            paddingRight: '10px',
+            paddingBottom: '16px'
+          }}
+        >
+          <Popover.Pane fixed>
+            <InlineGrid
+              columns={{ xs: '1fr', mdDown: '1fr', md: 'max-content max-content' }}
+              gap={0}
+            >
+              <Box
```

Comment:
There is some shifting happening in the UI when selecting between different sized options since the `OptionList` adds the checkmark to the selected option. Maybe see if you can fix the width to the largest element or something and account for the extra space needed for the checkmark.

https://github.com/RAAbbott/polaris-components/assets/41922804/455dc579-9403-4e47-9f3e-88f5077a80dd

Reply:
Good catch !